Lib/test/test_capi.py

```diff
@@ -333,6 +334,27 @@ def test_negative_refcount(self):
                          br'_Py_NegativeRefcount: Assertion ".*" failed; '
                          br'object has negative ref count')
 
+    @unittest.skipUnless(hasattr(_testcapi, 'structseq_newtype_doesnt_leak'),
+                         'need _testcapi.structseq_newtype_doesnt_leak')
+    def test_structseq_newtype_doesnt_leak(self):
+        # Warmup the caches
```

Comment:
Why are you reimplementing the refleak-detection machinery that already exists in libregrtest?

Reply:
I suspect the simple answer is that he didn't realize it exists.  Look at the command line arguments for the CPython test suite (python -m test -h).  Probably -R would handle this case.